handler/src/test/java/io/netty/handler/ssl/ApplicationProtocolNegotiationHandlerTest.java

```diff
@@ -43,6 +43,33 @@
 
 public class ApplicationProtocolNegotiationHandlerTest {
 
+    @Test
+    public void testRemoveItselfIfNoSslHandlerPresent() throws NoSuchAlgorithmException {
```

Comment:
Can you also add a test where the `SslHandler` addition is delayed post `ALPNHandler#handlerAdded()`?

Reply:
done ... PTAL again